src/test/java/petproject/controller/CarControllerTest.java

```diff
@@ -0,0 +1,204 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CarControllerTest {
+    protected static MockMvc mockMvc;
+    private static final String DELETE_FROM_CARS = "classpath:databases/delete-from-cars.sql";
+    private static final String ADD_THREE_CARS = "classpath:databases/add-three-cars.sql";
+    @Autowired
+    private ObjectMapper objectMapper;
+    @Autowired
+    private CustomUserDetailService customUserDetailService;
+    @MockBean
+    private TelegramNotificationsService notificationsService;
+    @MockBean
+    private UserRepository userRepositoryMockBean;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("databases/delete-from-cars.sql"));
+        }
+    }
+
+    @Test
+    @WithMockUser(username = "admin", roles = {"MANAGER"})
+    @Sql(scripts = DELETE_FROM_CARS,
+            executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = DELETE_FROM_CARS,
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName("Create a new car")
+    void createBook_validData_Success() throws Exception {
+        CreateCarDto carToRequest = ControllerTestUtil.createOneCarToRequest();
+        CarDto expected = ControllerTestUtil.createCarDto();
+
+        User testUser = ControllerTestUtil.createUserForTests();
+        when(userRepositoryMockBean.findByEmail(anyString()))
+                .thenReturn(Optional.of(testUser));
+
+        String jsonRequest = objectMapper.writeValueAsString(carToRequest);
+
+        MvcResult result = mockMvc.perform(
+                        post("/cars")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        verify(notificationsService, times(1))
+                .sendMessageOfCreateNewCar(any(User.class), any(Car.class));
+
+        CarDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), CarDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
```

Comment:
for other places also
```suggestion
        assertTrue(EqualsBuilder.reflectionEquals(expected, actual, "id"));
```